src/actions/minting/isNetworkDomainAvailable.ts

```diff
@@ -0,0 +1,25 @@
+import { domainNameToId, getLogger } from "../../utilities";
+import { ZNSHub } from "../../contracts/types";
+import { znsApiClient } from "../../api";
+
+const logger = getLogger("actions:getDomainMetadata");
+
+export const isNetworkDomainAvailable = async (
+  name: string,
+  hub: ZNSHub,
+  apiClient: znsApiClient
+): Promise<boolean> => {
+  logger.trace(`Checking network domain availability for: ${name}`);
+  // Check if name passes validation
+  const moderation = await apiClient.checkContentModeration(name);
+  if (moderation.flagged) {
+    logger.trace(
+      `${name}: was flagged for review, with reason: ${moderation?.reason}`
+    );
+    return false;
```

Comment:
Rather than throwing an error, I think eventually it should return a more detailed response indicating the failure status. I want to keep the integrity of this method as a simple true/false check to see if a domain can be registered